core/lib/include/kernel_defines.h

```diff
@@ -280,6 +280,28 @@ extern "C" {
 
 /* end{code-style-ignore} */
 
+/** @brief Declare an item as deprecated
+ *
+ * On modern (C2x) compilers, this expands to the C++11 style attribute
+ * `[[deprecated(reason)]]`; otherwise to the GNUism
+ * `__((attribute(deprecated))`, or is ignored.
```

Comment:
```suggestion
 * `__attribute_((deprecated))`, or is ignored.
```

Reply:
should be even `__attribute__`, shouldn't it?